scala/private/common.bzl

```diff
@@ -148,3 +157,16 @@ def create_java_provider(scalaattr, transitive_compile_time_jars):
         ),
         transitive_runtime_jars = scalaattr.transitive_runtime_jars,
     )
+
+PlusOneDeps = provider(
```

Comment:
we often put providers in different files to deal with that old bug. Is that unneeded now?

Can we comment this provider a bit more so people can understand a bit more the purpose?

Reply:
Don’t know about the old bug but I’ll put it in a different file if that’s the common thing. I also thought about doing it but wasn’t sure.

I’ll also add comments